docs/modules/array.md

```diff
@@ -29,23 +106,162 @@ _Example: Create an array containing a random number of values, then count them_
+---------- | ------------- | ---------:
+`array`    | Array to pop an element from | Yes
 
-``` 
-Array.indexOf(array, searchvalue)
+_Example: Pop the element 'b' off ['a', 'b']_
+
+```
+{!docs/examples/modules/array_pop.juttle!}
+```
+
+## Array.push
+
+Add an element to the end of an array
+```
+Array.push(array, element)
 ```
 
 Parameter  |  Description  |  Required?
 ---------- | ------------- | ---------:
-`array`    |   The array to search  | Yes
-`searchvalue`  |  The value for which to search  |  Yes
+`array`    | Array to add `element` to | Yes
+`element`  | Element to add | Yes
 
-_Example: Search for a random integer in an array of random integers_
+_Example: Push `'pajama'` and `'potatoes'` into the array `['banana']`_
 
 ```
-{!docs/examples/modules/array_indexof.juttle!}
+{!docs/examples/modules/array_push.juttle!}
+```
+
+## Array.reverse
+
+Reverse the order of elements in an array
+```
+Array.reverse(array)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    | Array reverse | Yes
+
+_Example: Reverse ['a', 'b'], yielding ['b', 'a']_
+
+```
+{!docs/examples/modules/array_reverse.juttle!}
+```
+
+## Array.shift
+
+Removes and returns the first element of an array
+```
+Array.shift(array)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    | Array to take first element from | Yes
+
+_Example: Take the element `'a'` out of `['a', 'b']`_
+
+```
+{!docs/examples/modules/array_shift.juttle!}
+```
+
+## Array.slice
+
+Returns a subarray of the given array
+```
+Array.slice(array, start, end)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    | Array to return a slice of | Yes
+`start`    | Index of subarray start | No; defaults to 0
+`end`      | Index of subarray end | No; defaults to end of array
+
+_Example: Take the slice `['banana']` of the array `['banana', 'pajama']`_
+
+```
+{!docs/examples/modules/array_slice.juttle!}
+```
+
+## Array.sort
+
+Sort an array
+```
+Array.sort(array)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    | Array to sort | Yes
+
+_Example: Sort `['c', 'a', 'b']`, yielding `['a', 'b', 'c']`._
+
+```
+{!docs/examples/modules/array_sort.juttle!}
+```
+
+## Array.splice
+
+Removes and returns a subarray, and replaces the subarray with given elements in the input array
+```
+Array.splice(array, start, end, elements...)
+```
+
+Parameter  |  Description  |  Required?
+---------- | ------------- | ---------:
+`array`    | Array to splice | Yes
+`start`    | Start of subarray to replace | No
+`end`    | End of subarray | No
+`elements`    | List of elements to add | No
+
+_Example: Replace `'banana'` with `'potatoes'` and return `['banana']` given the array `['banana', 'pajama']`._
+
+```
+{!docs/examples/modules/array_splice.juttle!}
+```
+
+## Array.toString
+
+Returns a string representation of a given array.
```

Comment:
Is there a separator if the array has multiple elements? Can you make the example be ["hello", "world"] instead of just ["hello"]?